src/rest/impl/deliveryLoans.service.ts

```diff
@@ -65,20 +73,184 @@ import ApplicationRoles from "../application-roles";
         }
       }
 
-      await SapDeliveriesServiceImpl.createStockTransferToSap(
+      const stockTransfer = await this.createSapStockTransfer(
+        loans,
+        Number(businessPartnerCode),
+        Number(salesPersonCode),
         new Date(),
-        [ comment ],
-        contactSapId,
-        salesPersonCode,
-        loans
+        [ comment ]
       );
-      
-      res.status(200).send(loans);
+
+      res.status(200).send(this.translateStockTransfer(stockTransfer));
     } catch (error) {
         logReject(createStackedReject("Delivery loan could not be created", error), getLogger());
         this.sendInternalServerError(res, "Delivery loan could not be created");
     }
-  
+  }
+
+    /**
+   * Create SAP stock transfer
+   *
+   * @param loans loans
+   * @param businessPartnerCode business partner code
+   * @param salesPersonCode sales person code
+   * @param docDate doc date
+   * @param comments comments
+   * @return promise of successful creation
+   */
+  private createSapStockTransfer = async (
+    loans: DeliveryLoan[],
+    businessPartnerCode: number,
+    salesPersonCode: number,
+    docDate: Date,
+    comments: string[]
+  ): Promise<SapStockTransfer> => {
+    try {
+      if (!loans.length) {
+        throw new Error("No loans");
+      }
+
+      const stockTransferLines = this.translateLoans(loans);
+
+      if (stockTransferLines.length < 1) {
+        throw new Error(`Stock transfer lines could not be constructed from following loans: ${JSON.stringify(loans, null, 2)}`);
+      }
+
+      const stockTransfersApi = await ErpClient.getStockTransfersApi();
+      const result = await stockTransfersApi.createStockTransfer({
+        businessPartnerCode: businessPartnerCode,
+        docDate: moment(docDate).format("YYYY-MM-DD"),
+        fromWarehouse: "100",
+        toWarehouse: "100",
+        salesPersonCode: salesPersonCode,
+        lines: stockTransferLines,
+        comments: this.joinComments(comments)
+      });
+
+      return result.body;
+    } catch (error) {
+      return Promise.reject(
+        createStackedReject(
+          "Create SAP stock transfer request failed.",
+          error instanceof HttpError ?
+            new Error(`${error.message}: ${JSON.stringify(error.body) || ""}`) :
+            error
+        )
+      );
+    }
+  }
+
+  /**
+   * Translates SAP stock transfer to list of delivery loans
+   *
+   * @param stockTransfer stock transfer
+   */
+  private translateStockTransfer = (stockTransfer: SapStockTransfer) => {
+    const linesByItem = _.groupBy(stockTransfer.lines, line => line.itemCode);
+    const deliveryLoans: DeliveryLoan[] = [];
+
+    for (const itemCode in linesByItem) {
+      const lines = linesByItem[itemCode];
+      const item = this.getLoanItemFromCode(itemCode);
+
+      if (!item) continue;
```

Comment:
The stock transfer has already been successfully created with correct item codes, so this should never really happen and it would anyways be confusing if an error would be returned since the data is already created, but it should indeed be logged in case it somehow did happen.